x-pack/plugins/endpoint/server/routes/endpoints.ts

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { IRouter } from 'kibana/server';
+import { SearchResponse } from 'elasticsearch';
+import { schema } from '@kbn/config-schema';
+import { EndpointAppContext, EndpointData } from '../types';
+import { kibanaRequestToEndpointListQuery } from '../services/endpoint/endpoint_query_builders';
+
+interface HitSource {
+  _source: EndpointData;
+}
+
+export interface EndpointResultList {
+  // the endpoint restricted by the page size
+  endpoints: EndpointData[];
+  // the total number of unique endpoints in the index
+  total: number;
+  // the page size requested
+  request_page_size: number;
+  // the index requested
+  request_index: number;
+}
+
+export function registerEndpointRoutes(router: IRouter, endpointAppContext: EndpointAppContext) {
+  router.post(
+    {
+      path: '/api/endpoint/endpoints',
+      validate: {
+        body: schema.nullable(
+          schema.object({
+            paging_properties: schema.arrayOf(
```

Comment:
I was wondering why the `paging_properties` is an `Array<Object>`. Would it be simpler to define it as an object with shorter options names:

```javascript
{
    paging: {
        size: number,
        page: number
    }
}
```

Reply:
I made the change because for some reason the order of property mattered. So size will always have to come before page, else it would be ignored. I also wanted validation from the schema too. I gave it a shot and this was the compromise.

Reply:
Thanks @nnamdifrankie . 
That seems weird - is that perhaps an issue with the new platform that we're using? If you think it is, maybe report it to the Kibana platform team (slack channel `#kibana-platform` )

Reply:
I will.